### PR TITLE
Lazily initalise the HttpClient for the Bitbucket OAuth2ClientRegistry

### PR DESCRIPTION
**Pull request by mjcheetham**
Lazily acquire the `HttpClient` instance from the factory inside of the `OAuth2ClientRegistry`. There is a relatively large cost to creating a new `HttpClient` instance (especially the first; due to config reading) that is being paid each time we just register the entire Bitbucket host provider.

Whilst we are here also cleanup some of the styling/field naming in this class to be consistent with the rest of GCM.

This saves approximately 120ms of startup time.

### CODE REVIEW

File: src/shared/Atlassian.Bitbucket/OAuth2ClientRegistry.cs [hunk 1]
```diff
@@ -30,15 +27,20 @@ public BitbucketOAuth2Client Get(InputArguments input)
...
         }
 
-        public void Dispose()
+        protected override void ReleaseManagedResources()
         {
-            http.Dispose();
-            settings.Dispose();
-            cloudClient = null;
-            dataCenterClient = null;
+            _httpClient.Dispose();
```
Comment (vdye) [inline src/shared/Atlassian.Bitbucket/OAuth2ClientRegistry.cs:32]: Will this cause an exception if `_httpClient` is `null`?

**Review by vdye — APPROVED** [review]
Looks good! I had one question about `ReleaseManagedResources`, hopefully easy to fix or explain either way. 🚀

File: src/shared/Atlassian.Bitbucket/OAuth2ClientRegistry.cs (hunk 1 above)
Comment (mjcheetham) [reply src/shared/Atlassian.Bitbucket/OAuth2ClientRegistry.cs]: Yes! Thanks for catching. Fixed in next push.

**Review by vdye — APPROVED** [review]

**Comment by mminns** [comment]
👍

**Review by mminns — COMMENTED** [review]
Looks good to me, thanks.

**Pull request merged**